BrainPortal/app/controllers/nh_messages_controller.rb

```diff
@@ -0,0 +1,125 @@
+#
+# NeuroHub Project
+#
+# Copyright (C) 2020
+# The Royal Institution for the Advancement of Learning
+# McGill University
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# RESTful controller for managing Messages.
+class NhMessagesController < NeurohubApplicationController
+  Revision_info = CbrainFileRevision[__FILE__] #:nodoc:
+
+  include Pagy::Backend
+
+  before_action :login_required
+
+  # GET /messages
+  # GET /messages.xml
+  def index #:nodoc:
+    # keep separate
+    @messages        = neurohub_messages
+    @messages_count  = @messages.count
+    @read_count      = @messages.where(:user_id => current_user.id, :read => true).count
+    @unread_count    = @messages.where(:user_id => current_user.id, :read => false).count
+    @page, @per_page = pagination_check(@messages, :nh_messages)
+    @pagy, @messages = pagy(@messages, :items => @per_page)
+  end
+
+  def new #:nodoc:
+    @message        = Message.new # blank object for new() form.
+    @message.header = "A personal message from #{current_user.full_name || current_user.login}"
+    nh_projects     = find_nh_projects(current_user)
+    contacts        = nh_projects.map { |x| x.users }.flatten.uniq.map { |x| x.own_group }
+    @recepients     = (current_user.assignable_groups.order(:name) & nh_projects | contacts).sort_by { |g| g.name }
+  end
+
+  # POST /messages
+  # POST /messages.xml
+  def create #:nodoc:
+    @message              = Message.new(message_params)
+    @message.message_type = :communication
+    @message.sender_id    = current_user.id
+    [:header, :description, :variable_text].each do |section|
+      if @message.send(section).include? '['
+        txt = 'Square brackets are not supported. Please use curly or round brackets only.'
```

Comment:
Please put your error messages in validation code in the model; if you don't want to interfere with the CBRAIN code, at least just put this all in  a method that you invoke here. That way you can validate in update() too

Reply:
I will double check, maybe allow use square brackets, that was a kludge to disable markup abuse by regular users. In past I have situation when management would like to bend rules for few exceptional cases. For example, user has a strange email or nick name but if he is important client or an executive, rules can be bended.